ast/src/parsed/mod.rs

```diff
@@ -41,33 +41,43 @@ pub enum PilStatement<T> {
 }
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
-pub struct SelectedExpressions<T> {
-    pub selector: Option<Expression<T>>,
-    pub expressions: Vec<Expression<T>>,
+pub struct SelectedExpressions<T, Ref = ShiftedPolynomialReference<T>> {
+    pub selector: Option<Expression<T, Ref>>,
+    pub expressions: Vec<Expression<T, Ref>>,
+}
+
+impl<T, Ref> Default for SelectedExpressions<T, Ref> {
+    fn default() -> Self {
+        Self {
+            selector: Default::default(),
+            expressions: Default::default(),
+        }
+    }
 }
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
-pub enum Expression<T> {
+pub enum Expression<T, Ref = ShiftedPolynomialReference<T>> {
     /// Reference to a constant, "%ConstantName"
     Constant(String),
-    PolynomialReference(ShiftedPolynomialReference<T>),
```

Comment:
In the future, I would not want to make a syntactic distinction between the different kinds of references (to constants, to polynomials, to public inputs)